kasa/discover.py

```diff
@@ -313,7 +365,7 @@ async def discover(
         discovery_timeout=5,
         discovery_packets=3,
         interface=None,
-        on_unsupported=None,
+        on_unsupported: OnUnsupportedCallable | None = None,
```

Comment:
I'd move the unrelated typings improvements into a separate PR, but your call.

Reply:
Done